src/client/common/process/pythonExecutionFactory.ts

```
@@ -32,12 +32,15 @@ export class PythonExecutionFactory implements IPythonExecutionFactory {
         return new PythonExecutionService(this.serviceContainer, processService, pythonPath);
     }
     public async createActivatedEnvironment(options: ExecutionFactoryCreateWithEnvironmentOptions): Promise<IPythonExecutionService> {
-        const envVars = await this.activationHelper.getActivatedEnvironmentVariables(options.resource, options.interpreter);
+        let envVars = await this.activationHelper.getActivatedEnvironmentVariables(options.resource, options.interpreter);
         const hasEnvVars = envVars && Object.keys(envVars).length > 0;
         sendTelemetryEvent(EventName.PYTHON_INTERPRETER_ACTIVATION_ENVIRONMENT_VARIABLES, undefined, { hasEnvVars });
         if (!hasEnvVars) {
             return this.create({ resource: options.resource, pythonPath: options.interpreter ? options.interpreter.path : undefined });
         }
+
+        // Not activating, just use current environment.
+        envVars = process.env;
```

Doesn't the above check in line `38` make this unnecessary?

---

No? envVars comes out as empty. Not sure how the !hasEnvVars makes this unnecessary. This is necessary so that even if the activate fails, it still uses the current environment, instead of an empty environment.

---

Although it seems like this should do that anyway because of the code in proc.ts. Maybe this change isn't necessary.

---

I'll change it to actually just not pass vars if it doesn't have them. Seems clearer.